src/components/AssetSelection/components/FiatMenuButton.tsx

```diff
@@ -5,32 +5,37 @@ import { useCallback, useMemo } from 'react'
 import { AssetRowLoading } from './AssetRowLoading'
 
 import { LazyLoadAvatar } from '@/components/LazyLoadAvatar'
+import type { FiatCurrencyItem } from '@/components/Modals/FiatRamps/config'
 import { getFiatFlagUrl } from '@/constants/fiatLogos'
-import type { FiatTypeEnumWithoutCryptos } from '@/constants/fiats'
+
 export type FiatMenuButtonProps = {
-  fiat: FiatTypeEnumWithoutCryptos
-  onFiatClick?: (fiat: FiatTypeEnumWithoutCryptos) => void
+  selectedFiatCurrency: FiatCurrencyItem
+  onClick?: () => void
   isDisabled?: boolean
   buttonProps?: ButtonProps
   isLoading?: boolean
 }
 
 export const FiatMenuButton = ({
-  fiat,
-  onFiatClick,
+  selectedFiatCurrency,
+  onClick,
   isDisabled,
   buttonProps,
   isLoading,
 }: FiatMenuButtonProps) => {
+  const flagUrl = useMemo(() => {
+    return getFiatFlagUrl(selectedFiatCurrency)
```

Comment:
nitpick: a bunch of these across the board - could be made a mapping instead of a const - then wouldn't even need to memo it, since that would be a lookup

Reply:
Done in [622b3b2](https://github.com/shapeshift/web/pull/10592/commits/622b3b2a98efda4bd32ce27229e2d08fcceff629)